packages/synapse-react-client/src/components/entity/page/title_bar/useGetEntityTitleBarProperties.tsx

```diff
@@ -42,18 +45,42 @@ export function useGetEntityTitleBarProperties(
 ): EntityProperty[] {
   const { data: bundle } = useGetEntityBundle(entityId, versionNumber)
 
+  const isContainer = !!(
+    bundle?.entityType && isContainerType(bundle.entityType)
+  )
+  const dataFileHandle = bundle && getDataFileHandle(bundle)
+  const parentId = bundle?.entity?.parentId
+  const storageLocationId = dataFileHandle?.storageLocationId
+
   const { data: entityChildrenResponse } = useGetEntityChildren(
     {
       parentId: entityId,
       includeTypes: Object.values(EntityType),
       includeTotalChildCount: true,
     },
-    { enabled: !!(bundle?.entityType && isContainerType(bundle.entityType)) },
+    { enabled: isContainer },
   )
 
-  const { data: uploadDestination } = useGetDefaultUploadDestination(entityId, {
-    enabled: !!(bundle?.entityType && isContainerType(bundle.entityType)),
-  })
+  const { data: defaultUploadDestination } = useGetDefaultUploadDestination(
+    entityId,
+    { enabled: isContainer },
+  )
+
+  // Note: defaultUploadDestination can't be used for non-container entities
+  // since the service checks that any entity in the path to the root has some
+  // project setting with a storage location, but does not look at the file entity's
+  // storage location id. Since a user can upload a file in any storage location
+  // and put it anywhere in the hierarchy, the file entity storage location id
+  // is where the file handle was actually uploaded. We must use a service that
+  // takes the storage location id in the input to get the storage location for a
+  // file entity.
+  const { data: storageLocationUploadDestination } =
+    useGetUploadDestinationForStorageLocation(parentId!, storageLocationId!, {
+      enabled:
+        !isContainer &&
+        parentId !== undefined &&
+        storageLocationId !== undefined,
+    })
```

Comment:
Thank you for this clarifying comment! What do you think about encapsulating both of these calls into a single `useGetUploadDestination` hook that calls the correct hook based on whether the entity is a container? Or maybe it makes more sense to keep them separate like this, since they really are fetching different things? I have no strong preference

Reply:
Oh that's a good idea! I'll make that change

Reply:
Actually, on second thought, I might keep them separate, since they are fetching different things -- for example, in `MetadataTable`, we get the storage location for file handles, but don't use the default upload destination for container entities, so we only need to call `useGetUploadDestinationForStorageLocation`. Keeping the hooks separate more clearly delineates which service will be called. What do you think?